src/snark0.ml

```diff
@@ -460,6 +460,46 @@ struct
             in
             res
 
+      module Array = struct
+        let num_true (bs : var array) =
+          Array.fold bs ~init:(Cvar.constant Field.zero) ~f:(fun x y ->
+              Cvar.add x (y :> Cvar.t) )
+
+        let any = function
```

Comment:
Maybe it's too much abstraction but we could implement both of these with one higher-order function :)